src/librustc_trans/back/linker.rs

```diff
@@ -199,6 +200,24 @@ impl<'a> Linker for GnuLinker<'a> {
     fn export_symbols(&mut self, _: &Session, _: &CrateTranslation, _: &Path) {
         // noop, visibility in object files takes care of this
     }
+
+    fn try_gold_linker(&mut self) {
+        let gold_exists = Path::new("/usr/bin/ld.gold").exists();
```

Comment:
No. No. No. You should check for it in all pathes listed in `$PATH`, not hard-code `/usr/bin`.